lib/services/redfish-validator-service.js

```diff
@@ -208,5 +212,40 @@ function redfishValidatorFactory(
 
     };
 
+    /**
+    * Get vendor name by id
+    * @param {String} id: nodeId
+    * @return {Object}: Object contains node vendor name and node info retrieved from database.
+    */
+    RedfishValidator.prototype.getVendorNameById = function(id) {
+        var result = {node: undefined, vendor: undefined};
+        return waterline.nodes.getNodeById(id)
+        .then(function(node){
+            if(!node){
+                throw new Errors.NotFoundError('invalid node id.');
+            }
+            result.node = node;
+            result.vendor = _getVendorNameByIdentifiers(node.identifiers);
+            return result;
+        });
+    };
+
+    function _getVendorNameByIdentifiers(_identifiers){
+        var vendor;
+        var ucsDnFormat = /^(sys|org-root)(\/[\w-]+)+$/;
+        var dellFormat = /^[0-9|A-Z]{7}$/;
+        _.forEach(_identifiers, function(_identifier){
+            var items = _identifier.split(':');
+            if (dellFormat.test(_identifier)){
+                vendor = 'Dell';
+                return false;
+            } else if (validator.isIP(items[0]) && ucsDnFormat.test(items[1])) {
+                vendor = 'Cisco';
+                return false;
+            }
+        });
+        return vendor;
+    }
+
```

Comment:
It might be a little confuse to add these common functions in redfish validator. What about changing the name of current service, or move these newly-added into a new service?